server/server.go

```diff
@@ -829,6 +829,15 @@ func (server *Server) loadConfig(configurations configs, globalConfiguration Glo
 						}
 						negroni.Use(cbreaker)
 					} else {
+						if frontend.Headers.HasCustomHeadersDefined() {
```

Comment:
@dtomcej Not sure to understand why this code block is inside the Circuit Breaker `else`. Don't you think you should move it before?